### PR TITLE
Add benchmarks

### PR DESCRIPTION
**Pull request by tj**
RE #79 

Hopefully not misusing the API in the bench. Output on my MBP illustrating the comparison with msgp, even without decoding any fields from the capn msg. 

```
BenchmarkUnmarshal-8             	 2000000	       777 ns/op	     208 B/op	       3 allocs/op
BenchmarkUnmarshal_Reuse-8       	 5000000	       282 ns/op	       0 B/op	       0 allocs/op
BenchmarkUnmarshalMsgp-8         	 5000000	       281 ns/op	      32 B/op	       2 allocs/op
BenchmarkUnmarshalMsgpReader-8   	     500	   3299178 ns/op	 294.01 MB/s	  322376 B/op	   20004 allocs/op
BenchmarkUnmarshalDecoder-8      	     300	   5520729 ns/op	 173.89 MB/s	 3200052 B/op	   60002 allocs/op
```

Hopefully I'll have some time this week to check out a solution if you don't get to it first! Looks like it should be comparable if all else is equal.

### CODE REVIEW

File: msgp_bench_test.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+type Event struct {
+	Name     string
+	BirthDay time.Time
+	Phone    string
+	Siblings int
+	Spouse   bool
+	Money    float64
+}
+
+func BenchmarkUnmarshalMsgp(b *testing.B) {
```
Comment (zombiezen) [inline msgp_bench_test.go]: Mind putting the MsgPack-related files and benchmarks behind a build tag?  I don't want the tests to depend on MsgPack unless specifically requested.

File: msgp_bench_test.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+		for {
+			var e Event
+			err := e.DecodeMsg(r)
+
+			if err == io.EOF {
+				break
+			}
+
+			if err != nil {
+				log.Fatal(err)
```
Comment (zombiezen) [inline msgp_bench_test.go]: Use `b.Fatal` instead (here and elsewhere).

File: msgp_bench_test.go [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+			}
+		}
+	}
+}
+
+func BenchmarkUnmarshalDecoder(b *testing.B) {
```
Comment (zombiezen) [inline msgp_bench_test.go]: I'd rename this to BenchmarkDecode.

File: msgp_bench_test.go [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+	for i := 0; i < b.N; i++ {
+		dec := capnp.NewDecoder(bytes.NewReader(blob))
+
+		for {
+			msg, err := dec.Decode()
+
+			if err == io.EOF {
+				break
+			}
```
Comment (zombiezen) [inline msgp_bench_test.go]: I'd also add an error check here for non-`io.EOF` errors.

**Comment by zombiezen** [comment]
Yeah, those numbers are not what I'd expect at all.  Good find!

**Comment by tj** [comment]
think I got everything, LMK if you want the capn benchmarks in a different file to just isolate the msgp ones

**Comment by zombiezen** [comment]
For now, let's just use `+build msgpbench`.

File: msgp_bench_test.go [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+
+			if err != nil {
+				b.Fatal(err)
+			}
+		}
+	}
+}
+
+func BenchmarkDecode(b *testing.B) {
```
Comment (zombiezen) [inline msgp_bench_test.go]: Let's pull out BenchmarkDecode into integration_test.go.

**Pull request merged**